fs/fs.go

```diff
@@ -81,13 +82,66 @@ var (
 	fusermountBin               = "fusermount"
 )
 
+type layerQueue struct {
+	mu       sync.Mutex
+	queue    chan func() string
+	cache    *hashset.Set
+	maxSize  int
+	currSize int
+}
+
+func (lq *layerQueue) Start() error {
+	if lq.queue != nil {
```

Comment:
nit: A common way to solve this is to have a `newLayerQueue` function which initializes the struture appropriately.